ProcessSlave.php

```diff
@@ -332,46 +333,48 @@ public function commandBootstrap(array $data, Connection $conn)
     /**
      * Handles incoming requests and transforms a $request into a $response by reference.
      *
-     * @param \React\Http\Request $request
-     * @param HttpResponse        $response
+     * @param RequestInterface $request
      *
+     * @return ResponseInterface
      * @throws \Exception
      */
-    public function onRequest(\React\Http\Request $request, HttpResponse $response)
+    public function onRequest(RequestInterface $request)
     {
         $this->prepareEnvironment($request);
 
+        $response = $this->handleRequest($request);
+
         if ($this->isLogging()) {
-            $this->setupResponseLogging($request, $response);
+            $this->logResponse($request, $response);
         }
-
-        $this->handleRequest($request, $response);
+        return $response;
     }
 
     /**
      * Handle a redirected request from master.
      *
-     * @param \React\Http\Request $request
-     * @param HttpResponse $response
+     * @param RequestInterface $request
+     * @return ResponseInterface
      */
-    protected function handleRequest(\React\Http\Request $request, HttpResponse $response)
+    protected function handleRequest(RequestInterface $request)
     {
         if ($bridge = $this->getBridge()) {
 
             if ($this->isServingStatic()) {
```

Comment:
We'll need to change `static` config to `static-directory`, then it should be possible. If new config will be empty -> no static, if not it will serve as static root dir. Is this what You meant?